activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1998,6 +1998,12 @@ def column_references(order_args)
             arg
           when Hash
             arg.keys.select { |e| e.is_a?(String) || e.is_a?(Symbol) }
+          when Arel::Attribute
+            "#{arg.relation.name}.#{arg.name}"
+          when Arel::Nodes::Ordering
+            if arg.relation.is_a?(Arel::Attribute)
```

Comment:
This feels like we're smashing through the Arel tree structure rather than treating it _as_ a tree... plus the stringifying just to then parse and de-string :confused:

Reply:
@matthewd I have refactored to no longer stringify/parse and instead handle each case directly.